xdsl/traits.py

```diff
@@ -203,30 +203,55 @@ class SymbolOpInterface(OpTrait):
     a `SymbolTable` (TODO). A Symbol operation should use the SymbolOpInterface interface to
     provide the necessary verification and accessors.
 
-    Currently the only requirement is a "sym_name" attribute of type StringAttr.
+    A Symbol operation may be optional or not. If - the default - it is not optional,
+    a `sym_name` attribute of type StringAttr is required. If it is optional,
+    the attribute is optional too.
 
-    Please see MLIR documentation for Symbol and SymbolTable for the requirements that are
-    upcoming in xDSL.
+    xDSL offers OptionalSymbolOpInterface as an always-optional SymbolOpInterface helper.
+
+    More requirements are defined in MLIR; Please see MLIR documentation for Symbol and
+    SymbolTable for the requirements that are upcoming in xDSL.
 
     https://mlir.llvm.org/docs/SymbolsAndSymbolTables/#symbol
     """
 
-    @staticmethod
-    def get_sym_attr_name(op: Operation) -> StringAttr:
+    def get_sym_attr_name(self, op: Operation) -> StringAttr | None:
         """
-        Returns the symbol of the operation
+        Returns the symbol of the operation, if any
         """
         # import builtin here to avoid circular import
         from xdsl.dialects.builtin import StringAttr
 
+        if "sym_name" not in op.attributes:
+            if self.is_optional_symbol(op):
+                return None
+            else:
+                raise VerifyException(
+                    f'Operation {op.name} must have a "sym_name" attribute of type '
+                    f"`StringAttr` to conform to {SymbolOpInterface.__name__}"
+                )
         attr = op.attributes["sym_name"]
-        assert isinstance(attr, StringAttr)
+        if not isinstance(attr, StringAttr):
+            raise VerifyException(
+                f'Operation {op.name} must have a "sym_name" attribute of type '
+                f"`StringAttr` to conform to {SymbolOpInterface.__name__}"
+            )
```

Comment:
Maybe you can merge this error with the previous one?
Just to not repeat the call.

Reply:
Okay so I'm rolling back on the condition :upside_down_face: Maybe My boolean-jitsu just got weak, not sure

Reply:
Okay, that was answer B :smiling_face_with_tear: